### PR TITLE
Add uvbybeta function

### PR DESCRIPTION
**Pull request by TestSubjector**
Slightly longer function than the usual :wink: 

WIP, because of the many conditional statements that need to be covered and their values compared with that of IDL for correctness, by tests. (I estimate that this function will require at least 5 times the tests compared any other function in AstroLib.jl)

- [x]  Add tests for `n`=1 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=2 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=3 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=4 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=5 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=6 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=7 and compare values with that of IDL AstroLib
- [x]  Add tests for `n`=8 and compare values with that of IDL AstroLib
- [x]  Make return type stable

### CODE REVIEW

File: src/uvbybeta.jl [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        else
+            eby_in = (0.5126 * by - c1 - 0.3645) / 0.3226
+            by0 = by - eby_in
+
+            if by0 < 0.79
+                eby_in = (-0.7875 * by - c1 + 0.6585) / -0.9775
+            end
+            by0 = by - eby_in
+
+            # This will never happen (unless both by and c1 are <0, but is
```
Comment (TestSubjector) [inline src/uvbybeta.jl]: Need a review of this. I'm assuming that Stromgren b-y color `by` and Stromgren Balmer discontinuity parameter `c1` can't be negative. If that is true, then this condition is never satisfied.

Comment (giordano) [inline src/uvbybeta.jl]: The color index may be negative, I'm not sure what the discontinuity parameter is.  I'd be safe and keep this test.

Comment (TestSubjector) [inline src/uvbybeta.jl]: So after doing the math properly, it seems that the condition for that if statement will never be true.

Comment (giordano) [inline src/uvbybeta.jl]: Could you please elaborate why?

Comment (TestSubjector) [inline src/uvbybeta.jl]: Well, going through the outer else statement - The first time `by0` is calculated, it has to be less than 0.79 and then after its recalculated, it has to be less than 0.65. Essentially, two inequalities have to be solved,

by - ((0.5126 * by - c1 - 0.3645) / 0.3226) < 0.79 and by - ((-0.7875 * by - c1 + 0.6585) / -0.9775) < 0.65.

This has no [common solution for `by` and `c1`](http://www.wolframalpha.com/input/?i=x+-+((0.5126+*+x+-+y+-+0.3645)+%2F+0.3226)+%3C+0.79+%26+x+%2B+((-0.7875+*+x+-+y+%2B+0.6585)+%2F+0.9775)+%3C+0.65)

File: src/uvbybeta.jl [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+end
+
+"""
+    uvbybeta(by, m1, c1, n[, hbeta=NaN, eby_in=NaN]) -> te, mv, eby, delm0, radius
+
+### Purpose ###
+
+Derive dereddened colors, metallicity, and Teff from Stromgren colors.
+
+### Explanation ###
```
Comment (giordano) [inline src/uvbybeta.jl]: This section can be removed

**Pull request merged**